coderd/database/dbauthz/querier.go

```diff
@@ -1697,6 +1697,10 @@ func (q *querier) GetWorkspaceProxyByID(ctx context.Context, id uuid.UUID) (data
 	return fetch(q.log, q.auth, q.db.GetWorkspaceProxyByID)(ctx, id)
 }
 
+func (q *querier) GetWorkspaceProxyByName(ctx context.Context, name string) (database.WorkspaceProxy, error) {
```

Comment:
Is name currently unique?

Reply:
Yes:

https://github.com/coder/coder/blob/dc44a71171175322216bc34714ce0e42982f4bea/coderd/database/dump.sql#L892-L892

Reply:
Just realized it should probably be `lower(name)`

Reply:
Can be a follow up PR